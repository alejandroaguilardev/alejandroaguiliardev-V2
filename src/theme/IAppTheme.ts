export interface IAppTheme {
    children:  React.ReactElement | React.ReactElement[] | React.ReactNode;
}