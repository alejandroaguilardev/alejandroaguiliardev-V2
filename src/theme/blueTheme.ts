import { createTheme } from "@mui/material";
import { red } from "@mui/material/colors";

export const blueTheme = createTheme({
	palette: {
		// mode:'dark'
		primary: {
			main: "#0052B4",
			light: "#fff",
			dark: "#09162e",
			contrastText: "#fff",
		},
		secondary: {
			main: "#2946F7",
			// main: "#023F89",
		},
		background: {
			default: "#fbfcfc",
			paper:'#fff'
		},
		text: {
			primary: "#222",
			secondary: "#444",
			//   disabled: string;
		},
		error: {
			main: red.A400,
		},
	},
});
