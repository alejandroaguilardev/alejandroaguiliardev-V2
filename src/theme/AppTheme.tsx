import { ThemeProvider } from "@emotion/react";
import { CssBaseline } from "@mui/material";
import { IAppTheme } from "./IAppTheme";
import { useThemeContext } from "../context";
import { blueTheme } from "./blueTheme";
import { darkTheme } from "./darkTheme";
import { useMemo } from "react";

export const AppTheme = ({ children }: IAppTheme) => {
	const { theme } = useThemeContext();

	const themeSelected = useMemo(
		() => (theme.mode === "dark" ? darkTheme : blueTheme),
		[theme]
	);

	return (
		<ThemeProvider theme={themeSelected}>
			<CssBaseline />
			{children}
		</ThemeProvider>
	);
};
