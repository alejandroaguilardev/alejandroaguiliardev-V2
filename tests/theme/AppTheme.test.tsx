import React from "react";
import { render, screen } from '@testing-library/react';
import { AppTheme } from "../../src/theme/AppTheme";



describe("AppTheme test", () => {

	test("AppTheme SnapShots", () => {
		const { container } = render(
			<AppTheme>
				<h1>AppTheme</h1>
			</AppTheme>
		);

		expect(container).toMatchSnapshot();
		expect(screen.getByText("AppTheme")).toBeTruthy(); 
	});
});
