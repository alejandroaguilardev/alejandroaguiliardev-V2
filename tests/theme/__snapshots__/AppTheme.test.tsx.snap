// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTheme test AppTheme SnapShots 1`] = `
<div>
  <h1>
    AppTheme
  </h1>
</div>
`;
